src/components/Card.js

```diff
@@ -1,21 +1,34 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
-    this._name = name;
-    this._link = link;
+  constructor({
+    cardData,
+    cardSelector,
+    handleDeleteClick,
+    handleImageClick,
+    api,
+  }) {
+    this._name = cardData.name;
+    this._link = cardData.link;
+    this._cardId = cardData._id;
+    this._api = api;
     this._cardSelector = cardSelector;
+    this._handleDeleteClick = handleDeleteClick;
     this._handleImageClick = handleImageClick;
   }
 
   _setEventListeners() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeButton();
-      });
+    this._likeButton.classList.toggle(
+      "card__like-button_active",
+      this._isLiked
```

Comment:
* New cards appear already `liked`, but they should not be


`_isLiked` doesn't exist. Please, define it first


```js
    this._handleImageClick = handleImageClick;
    this._isLiked = cardData.isLiked;  // define
  }

  _setEventListeners() {
    this._likeButton.classList.toggle(
      "card__like-button_active",
      this._isLiked  // use
    );
```